### PR TITLE
You see me rollin' - Roller Bed Tweaks

### PR DESCRIPTION
**Pull request by 1Sparky1**
A full list of the changes, complete with better explanations and links to the original ideas can be found [here](https://forums.aurorastation.org/topic/15543-roller-bed-tweaks/?tab=comments#comment-145154).

In Summary:
Resprites roller beds.
Swaps the IV side.
Allows body bags to be buckled to them.
Makes them normal sized.
Allows you to ctrl-click the patient to pull the roller.

changes:
  - tweak: "Changed the roller bed sprites. They are now a bit bulkier (for ease of clicking) and a bit brighter."
  - tweak: "Roller bed IVs now show up on the left of the bed, not the right, so they are no longer blocked by medhuds. The fluid low light is now a tad more obvious too."
  - tweak: "You can now buckle body bags and cryo bags to roller beds."
  - tweak: "Roller beds are now normal sized items, so they fit in your bag."
  - tweak: "Trying to pull someone buckled to something now attempts to pull that thing instead."

### CODE REVIEW

**Comment by 1Sparky1** [comment]
!review

**Review by OneOneThreeEight — APPROVED** [review]

File: code/game/objects/buckling.dm [hunk 1]
```diff
@@ -1,103 +1,118 @@
 /obj
 	var/can_buckle = 0
+	var/can_buckle_bags = 0
```
Comment (NonQueueingMatt) [inline code/game/objects/buckling.dm:3]: This mechanic shouldn't be specific to bags if implemented. Objects should be able to decide what exactly can be buckled to them via either a list or a proc that returns a list of objects.

File: code/game/objects/buckling.dm [hunk 2]
```diff
@@ -1,103 +1,118 @@
...
 	var/buckle_movable = 0
 	var/buckle_dir = 0
 	var/buckle_lying = -1 //bed-like behavior, forces mob.lying = buckle_lying if != -1
 	var/buckle_require_restraints = 0 //require people to be handcuffed before being able to buckle. eg: pipes
 	var/mob/living/buckled_mob = null
+	var/obj/structure/closet/body_bag/buckled_bag = null
```
Comment (NonQueueingMatt) [inline code/game/objects/buckling.dm:9]: Similarly, this variable should then be only an /atom/movable or /obj variable. Though I guess it's possible to unify this with the buckled_mob variable and only have `var/atom/movable/buckled`?

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/game/objects/buckling.dm (hunk 1 above)
Comment (1Sparky1) [reply code/game/objects/buckling.dm:3]: I knew I missed something in my clean-up. I meant to entirely unify the buckling into one thing, but obviously missed this.

File: code/game/machinery/rechargestation.dm [hunk 3]
```diff
@@ -284,7 +284,7 @@
 				user.visible_message(SPAN_NOTICE("After a great effort, [user] manages to get [C] into the recharging unit!"))
 				return 1
 			else
-				to_chat(user, SPAN_DANGER("Failed loading [C] into the charger. Please ensure that [C] has a power cell and is not buckled down, and that the charger is functioning."))
+				to_chat(user, SPAN_DANGER("Failed loading [C] into the charger. Please ensure that [C] has a power cell and is not buckled_to down, and that the charger is functioning."))
```
Comment (Geevies) [inline code/game/machinery/rechargestation.dm:287]: replace all error

File: code/game/objects/buckling.dm [hunk 4]
```diff
@@ -1,103 +1,112 @@
 /obj
-	var/can_buckle = 0
+	var/list/can_buckle = list()
```
Comment (Geevies) [inline code/game/objects/buckling.dm:2]: make this into a lazylist instead

File: code/modules/mob/living/carbon/human/human_attackhand.dm [hunk 5]
```diff
@@ -137,8 +137,8 @@
 				w_uniform.add_fingerprint(M)
 
 			var/obj/item/grab/G = new /obj/item/grab(M, src)
-			if(buckled)
-				to_chat(M, "<span class='notice'>You cannot grab [src], [src.get_pronoun("he")] [get_pronoun("is")] buckled in!</span>")
+			if(buckled_to)
+				to_chat(M, "<span class='notice'>You cannot grab [src], [src.get_pronoun("he")] [get_pronoun("is")] buckled_to in!</span>")
```
Comment (Geevies) [inline code/modules/mob/living/carbon/human/human_attackhand.dm:141]: replace all error

File: code/modules/mob/living/carbon/human/human_defense.dm [hunk 6]
```diff
@@ -559,8 +559,8 @@ emp_act
 		src.w_uniform.add_fingerprint(src)
 
 	var/obj/item/grab/G = new /obj/item/grab(user, src)
-	if(buckled)
-		to_chat(user, "<span class='notice'>You cannot grab [src], [get_pronoun("he")] [get_pronoun("is")] buckled in!</span>")
+	if(buckled_to)
+		to_chat(user, "<span class='notice'>You cannot grab [src], [get_pronoun("he")] [get_pronoun("is")] buckled_to in!</span>")
```
Comment (Geevies) [inline code/modules/mob/living/carbon/human/human_defense.dm:563]: replace all error

File: config/example/tips.txt [hunk 7]
```diff
@@ -10,7 +10,7 @@ Glass shards can be welded to make glass, and metal rods can be welded to make m
 If you need to drag multiple people either to safety or to space, bring a locker over and stuff them all in before hauling them off.
 You can grab someone by clicking on them with the grab intent, then upgrade the grab by clicking on the grab button in your active hand. An aggressive grab will allow you to place someone on a table by clicking on it, or throw them by toggling on throwing.
 Holding alt and left clicking a tile will allow you to see its contents in the top right window pane, which is much faster than right clicking.
-The resist button will allow you to resist out of handcuffs, being buckled to a chair or bed, out of locked lockers and more. Whenever you're stuck, try resisting!
+The resist button will allow you to resist out of handcuffs, being buckled_to to a chair or bed, out of locked lockers and more. Whenever you're stuck, try resisting!
```
Comment (Geevies) [inline config/example/tips.txt:13]: replace all error

File: html/changelog.html [hunk 8]
```diff
@@ -2625,7 +2625,7 @@ <h3 class="author">TheDocOct updated:</h3>
 			</ul>
 			<h3 class="author">mikomyazaki updated:</h3>
 			<ul class="changes bgimages16">
-				<li class="tweak">Closets, chairs, etc. without a mob inside/buckled will not cause a slowdown when being dragged.</li>
+				<li class="tweak">Closets, chairs, etc. without a mob inside/buckled_to will not cause a slowdown when being dragged.</li>
```
Comment (Geevies) [inline html/changelog.html:2628]: whole file is a replace all error

File: html/changelogs/.all_changelog.yml [hunk 9]
```diff
@@ -11236,7 +11236,7 @@ DO NOT EDIT THIS FILE BY HAND!  AUTOMATICALLY GENERATED BY ss13_genchangelog.py.
   - bugfix: Offworlder goggles will now apply the chosen loadout color properly.
   Geeves:
   - bugfix: Gemstones larger than small now properly spawn.
-  - bugfix: You can now flip your welding mask up and down even when buckled.
+  - bugfix: You can now flip your welding mask up and down even when buckled_to.
```
Comment (Geevies) [inline html/changelogs/.all_changelog.yml:11239]: whole file is a replace all error

**Review by Geevies — CHANGES_REQUESTED** [review]
good idea, but i think it'd be better to make buckling a component, i believe vorestation has something similar

File: code/game/objects/buckling.dm (hunk 4 above)
Comment (1Sparky1) [reply code/game/objects/buckling.dm:2]: Lazylist doesn't seem to like typepaths, so...

**Comment by Geevies** [comment]
when lazylisting 'em, you can just do `var/list/can_buckle` on the main /obj and assign lists as you do otherwise, initting them on initialize just undoes the lazylist

/obj
    var/list/can_buckle

/obj/structure/bed
    can_buckle = list(/mob/living)

**Review by skull132 — APPROVED** [review]

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**